src/main/java/net/shadowmage/ancientwarfare/npc/entity/faction/NpcFaction.java

```diff
@@ -249,6 +252,39 @@ public void onEntityUpdate() {
 
 	}
 
+	private String getEntitySound() {
+		if (getAdditionalAttributeValue(AdditionalAttributes.ENTITY_SOUND).isPresent()) {
+			return "entity_" + getAdditionalAttributeValue(AdditionalAttributes.ENTITY_SOUND).get();
+		} else
+			return "generic";
+	}
+
+	@Override
+	protected SoundEvent getHurtSound(DamageSource damageSourceIn) {
+		String sound = getEntitySound() + "_hurt";
+
+		if (!sound.equals("generic_hurt") && AWNPCSounds.isValidSound(sound)) {
+			return AWNPCSounds.getSoundEventFromString(sound);
+		} else
+			return SoundEvents.ENTITY_GENERIC_HURT;
+	}
+
+	@Override
+	protected SoundEvent getDeathSound() {
+		String sound = getEntitySound() + "_death";
+		if (!sound.equals("generic_death") && AWNPCSounds.isValidSound(sound)) {
+			return AWNPCSounds.getSoundEventFromString(sound);
+		} else
+			return SoundEvents.ENTITY_GENERIC_DEATH;
+	}
+
+	public void playAttackSound() {
+		String sound = getEntitySound() + "_attack";
+		if (!sound.equals("generic_attack") && AWNPCSounds.isValidSound(sound)) {
+			playSound(AWNPCSounds.getSoundEventFromString(sound), 1.0F, 1.2F / (rand.nextFloat() * 0.2F + 0.9F));

```

Comment:
The sound event should be cached in a field so that there isn't a need to query to compile its id and query for it every time a sound needs to be played. Probably best way to approach that is compile the name in constructor and cache it into something like private final SoundEvent attackSound;